server/src/main/java/com/testsigma/repository/TestStepRepository.java

```diff
@@ -129,4 +129,11 @@ List<TestStep> findAllByWorkspaceVersionIdAndNaturalTextActionId(@Param(value =
   @Modifying
   @Query("DELETE FROM TestStep step WHERE step.testCaseId = :testCaseId")
   void deleteStepsByTestCaseId(@Param("testCaseId") Long testCaseId);
+
+  List<TestStep> findAllByStepGroupId(Long testComponentId);
+
+  @Modifying
+  @Query("DELETE FROM TestStep step WHERE step.stepGroupId = :stepGroupId")
+  void deleteStepsByTestComponentId(@Param("stepGroupId") Long stepGroupId);
```

Comment:
@nixonakash01 You can use deleteAllByStepGroupId instead of using a query here.

Reply:
@tarun-testsigma deleteAllByStepGroupId is not present anywhere, thats why i added a query with function "deleteStepsByTestComponentId" in repository and called it in service. I checked its working.

deleteStepsByTestComponentId will be renamed to deleteStepsByStepGroupId as per OS, the change is be commited